### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by mstfygci**
Werewolf

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,20 @@
 import './App.css';
 import Three from './three';
+import 'bootstrap/dist/css/bootstrap.min.css'
+import 'bootstrap-icons/font/bootstrap-icons.min.css'
```
Comment (Fabsolute) [inline src/App.tsx]: These styles should be moved in `App.css` or `Layout` itself.

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-            <button style={{position: 'absolute', top: 300, left: 300}}>Oy</button>
-        </div>
-    );
+  return (
+    <div style={{position: 'relative'}}>
+      <Three/>
+      <Layout>
+        <Menu/>
+      </Layout>
+      {/*<button style={{position: 'absolute', top: 300, left: 300}}>Oy</button>*/}
```
Comment (Fabsolute) [inline src/App.tsx]: should be removed

File: src/components/Info.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {Button, ProgressBar} from 'react-bootstrap';
+
+export function Info() {
+  const levelStatus = 60;
+  const name = 'mstfygci';
+  const level = 13;
+  return (
+    <Button variant="light" className="mt-3 d-flex" style={{width: 'auto'}}>
```
Comment (Fabsolute) [inline src/components/Info.tsx]: we shouldnt use inline styles

File: src/components/Info.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      <i className="bi bi-person-circle me-3" style={{fontSize: '2rem'}}></i>
+      <div>
+        <div className="me-3">
+          {name}
+          <ProgressBar variant="info" now={levelStatus} label={`${levelStatus}%`}/>
```
Comment (Fabsolute) [inline src/components/Info.tsx]: we should use `percentage` instead of `levelStatus`

File: src/components/Menu.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import {Button, ButtonToolbar} from 'react-bootstrap';
+
+type ButtonType = {
+  title: string,
+  backgroundColor: string,
+  color: string
+}[]
```
Comment (Fabsolute) [inline src/components/Menu.tsx]: This type is not representing an array. So it shouldn't be an array.

File: src/components/Menu.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      backgroundColor: 'gainsboro',
+      color: 'black'
+    }
+  ]
+  return (
+    <ButtonToolbar
+      className="position-absolute d-flex row start-50 top-50 translate-middle justify-content-center pe-0">
+      {buttons.map((item, index) => {
+        return (
+          <div className={`row justify-content-center ${index != 0 && ' mt-2'}`}>
```
Comment (Fabsolute) [inline src/components/Menu.tsx]: key required

File: src/layout/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                        title={<i className="bi bi-person-hearts"></i>}
+                        variant="light"
+                        id="bg-nested-dropdown"
+        >
+          <Dropdown.Item eventKey="1">Dropdown link</Dropdown.Item>
+          <Dropdown.Item eventKey="2">Dropdown link</Dropdown.Item>
+        </DropdownButton>
+        <Button variant="light"><i className="bi bi-gear-fill"></i></Button>
+      </ButtonToolbar>
+      <Button variant="light" className="mt-3 d-flex" style={{width: 'auto'}}>
```
Comment (Fabsolute) [inline src/layout/Header.tsx]: We have info component. So, we shouldn't use the same codes here.

**Review by Fabsolute — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (mstfygci) [reply src/App.tsx]: moved to App.css

File: src/App.tsx (hunk 2 above)
Comment (mstfygci) [reply src/App.tsx]: Removed

File: src/components/Info.tsx (hunk 3 above)
Comment (mstfygci) [reply src/components/Info.tsx]: removed

File: src/components/Info.tsx (hunk 4 above)
Comment (mstfygci) [reply src/components/Info.tsx]: Changed

File: src/components/Menu.tsx (hunk 5 above)
Comment (mstfygci) [reply src/components/Menu.tsx]: Removed from type and added to the used place

File: src/components/Menu.tsx (hunk 6 above)
Comment (mstfygci) [reply src/components/Menu.tsx]: Added `key` property to ButtonType

Comment (Fabsolute) [reply src/components/Menu.tsx]: Div still creating without key.

File: src/layout/Header.tsx (hunk 7 above)
Comment (mstfygci) [reply src/layout/Header.tsx]: Replaced with `Info` component

File: src/App.tsx [hunk 8]
```diff
@@ -1,14 +1,17 @@
...
 function App() {
-    return (
-        <div style={{position: 'relative'}}>
-            <Three/>
-
-            <button style={{position: 'absolute', top: 300, left: 300}}>Oy</button>
-        </div>
-    );
+  return (
+    <div style={{position: 'relative'}}>
```
Comment (Fabsolute) [inline src/App.tsx]: inline style

File: src/components/Info.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import {Button, ProgressBar} from 'react-bootstrap';
+
+export function Info() {
+  const percentage = 60;
+  const name = 'mstfygci';
+  const level = 13;
+  return (
```
Comment (Fabsolute) [inline src/components/Info.tsx:8]: I think we should add space between codes and return.

File: src/components/Menu.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export function Menu() {
+  const buttons: ButtonType[] = [
+    {
+      key: 'play',
+      title: 'PLAY',
+      backgroundColor: 'gainsboro',
+      color: 'black'
+    }, {
```
Comment (Fabsolute) [inline src/components/Menu.tsx]:
```suggestion
    }, 
    {
```

File: src/components/Info.tsx [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        <div className="me-3">
+          {name}
+          <ProgressBar variant="info" now={percentage} label={`${percentage}%`}/>
+        </div>
+      </div>
+      <i className="bi bi-star-fill h2 text-warning">
+        {level}
+      </i>
+    </Button>
+  )
```
Comment (Fabsolute) [inline src/components/Info.tsx]:
```suggestion
  );
```

File: src/layout/index.tsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import {PropsWithChildren} from 'react';
+import {Footer} from './Footer.tsx';
+import {Header} from './Header.tsx';
+
+export function Layout({children}: PropsWithChildren) {
+    return <>
+        <Header/>
+        <main>{children}</main>
+        <Footer/>
+    </>
```
Comment (Fabsolute) [inline src/layout/index.tsx]:
```suggestion
    return (
    <>
        <Header/>
        <main>{children}</main>
        <Footer/>
    </>
    );
```

File: src/layout/Header.tsx [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import {Info} from '../components/Info.tsx';
+
+export function Header() {
+
```
Comment (Fabsolute) [inline src/layout/Header.tsx]: remove the empty line

**Review by Fabsolute — CHANGES_REQUESTED** [review]

**Comment by Fabsolute** [comment]
You don't have to reply to my comments if you are okay with it. Just press the resolve conversation button.

File: src/components/Menu.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import {Button, ButtonToolbar} from 'react-bootstrap';
+
+type ButtonType = {
+  title: string,
+  backgroundColor: string,
+  color: string
+}
```
Comment (Fabsolute) [inline src/components/Menu.tsx]: `;`

You should check all my comments for the all project. So, we shouldn't repeat ourselves.

File: src/components/Menu.tsx [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+      color: 'black'
+    }
+  ]
+
+  return (
+    <ButtonToolbar
+      className="position-absolute d-flex row start-50 top-50 translate-middle justify-content-center pe-0">
+      {buttons.map((item, index) => {
+        return (
+          <div className={`row justify-content-center ${index != 0 && ' mt-2'}`} key={`menu_${index}`}>
```
Comment (Fabsolute) [inline src/components/Menu.tsx]: Keys shouldn't created with indexes. It is against their nature.

**Review by Fabsolute — CHANGES_REQUESTED** [review]

File: src/components/Menu.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
+import {Button, ButtonToolbar} from 'react-bootstrap';
+
+type ButtonType = {
+  key: string,
+  title: string,
+  backgroundColor: string,
+  color: string
```
Comment (Fabsolute) [inline src/components/Menu.tsx:7]: use a semicolon instead of a comma.

**Pull request merged**